src/components/List.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { View, FlatList, Text, Image } from 'react-native';
+import Header from './Header';
+import ItemList from './ItemList';
+import addIcon from '../assets/icons/add.png';
+import Button from './Button';
+import styles from './List.styles';
+import strings from '../assets/strings';
+
+class List extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: props.items,
+    };
+  }
+
+  componentWillReceiveProps({ items }) {
+    this.setState({ ...this.state, items });
```

Comment:
in this kind of scenarios you should use the [Functional setState](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b)